api/src/main/java/org/openmrs/module/queue/api/QueueService.java

```diff
@@ -53,16 +58,19 @@ public interface QueueService {
 	 * @param queue the queue to be saved
 	 * @return saved {@link org.openmrs.module.queue.model.Queue}
 	 */
+	@Authorized({ PrivilegeConstants.ADD_QUEUES, PrivilegeConstants.EDIT_QUEUES })
```

Comment:
Could there be a reason why for this service you are doing `Add` and `Edit` and yet for others you are doing `Manage`?

Reply:
`saveQueue()` is used to both update and add new queues, so different privileges must be required, ergo the `Add` and `Edit`.

Reply:
but I have realized that `Manage` also provides both the `Add` and `Edit` privileges at once which is better, let me change that, thanks🙂